### PR TITLE
Allow cross version deployment of meshmodel components

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: Ashish Tiwari <ashishjaitiwari15112000@gmail.com>

**Notes for Reviewers**

This PR fixes:
Prior to this PR, a component in the design with a certain model version can only be deployed when the registry has the component for that model version. This PR lifts off this restriction.
If the version is available then it will be preferred other wise any available version will be used.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by Abhishek-kumar09 — COMMENTED** [review]
should there be an event "info" triggered in the case the cross version deployment is going on, just to make sure user knows about it.

**Review by leecalcote — DISMISSED** [review]
LGTM

File: server/models/pattern/resource/selector/workload.go [hunk 1]
```diff
@@ -29,15 +32,17 @@ func (s *Selector) Workload(name string, version string, model string) (v1alpha1
 				comp, ok = en.(v1alpha1.ComponentDefinition)
 				if ok {
 					found = true
+				}
+				if comp.Model.Version == version { //prefer to use the correct version, if available
```
Comment (leecalcote) [inline server/models/pattern/resource/selector/workload.go:36]: It makes sense to include a debug level log of the decision the engine makes of which registrant to use.

Comment (leecalcote) [reply server/models/pattern/resource/selector/workload.go:36]: After a log message is considered, this LGTM

**Comment by leecalcote** [comment]
@Revolyssup a merge conflict has emerged

**Comment by leecalcote** [comment]
@Revolyssup checking in. The log message and merge conflict linger.

File: server/models/pattern/resource/selector/workload.go (hunk 1 above)
Comment (Revolyssup) [reply server/models/pattern/resource/selector/workload.go:36]: Okay

**Pull request merged**